packages/hawtio/src/ui/page/HawtioPage.tsx

```diff
@@ -18,10 +18,12 @@ import { HawtioBackground } from './HawtioBackground'
 import { HawtioHeader } from './HawtioHeader'
 import { HawtioLoading } from './HawtioLoading'
 import { HawtioSidebar } from './HawtioSidebar'
+import { PluginNodeSelectionContext, usePluginNodeSelected } from '@hawtiosrc/plugins'
```

Comment:
hmm, with this dependency, now we have a circular dependency between the core ui part and the plugins. Is it what we really need?

Reply:
Not quite. The `PluginNodeSelectionContext is not in the JMX plugin but in the `plugin` module. I thought hard about this. If we want to move it to a core module then that should be ok I think.

Reply:
https://github.com/hawtio/hawtio-next/pull/115/files#diff-4b7e788f0291a4786b206a46509a458911884d644d2bc9e245bda4d449b0344f

Reply:
But the concept of `selectedNode` comes from JMX plugin's MBean tree, so that's why I think the page depends on JMX plugin.

Reply:
The MBean tree is no longer in the JMX plugin. It is fetched and provided by [workspace](https://github.com/hawtio/hawtio-next/blob/main/packages/hawtio/src/plugins/shared/workspace.ts#L14), which is in shared.

Reply:
Shared is still part of the plugins. I don't think it improved the situation. For now, let's accept the circular dependency and go ahead.